bc/sponsorship/models.py

```diff
@@ -0,0 +1,79 @@
+from django.db import models
+
+from bc.core.models import AbstractDateTimeModel
+from bc.users.models import User
+
+
+class Sponsorship(AbstractDateTimeModel):
+    user = models.ForeignKey(
+        User,
+        help_text="The user sponsoring the Bot",
+        related_name="sponsorships",
+        on_delete=models.CASCADE,
+    )
+    original_amount = models.DecimalField(
+        help_text="Initial amount of money given by the sponsor",
+        max_digits=10,
+        decimal_places=2,
+    )
+    current_amount = models.DecimalField(
+        help_text="Amount of money that's left to buy documents",
+        max_digits=10,
+        decimal_places=2,
+        default=0,
+    )
+
+    class Meta:
+        ordering = ["date_created"]
+
+    def __str__(self) -> str:
+        return f"{self.pk}: {self.user.username} - {self.original_amount}"
+
+
+class Transaction(AbstractDateTimeModel):
+    SPONSORSHIP = 1
+    DOCUMENT_PURCHASE = 2
+    ADJUSTMENT = 3
+    CHOICES = (
```

Comment:
CHOICES is a weird variable name. If you wind up using it in other code, you'd have:

```
Transaction.CHOICES
```

Not great. Perhaps:

```
Transactions.TYPES
```

That's dangerously close to a reserved word in Python, but frankly, I worry about your use of `type` as the field name too, but I think it should be fine.

Reply:
got it! Thanks for your suggestion.